python/beeai_framework/adapters/a2a/agents/agent.py

```diff
@@ -205,18 +199,22 @@ async def clone(self) -> "A2AAgent":
         return cloned
 
     def _convert_message_to_framework_message(
-        self, input: str | AnyMessage | a2a_types.Message | a2a_types.Artifact
+        self, input: str | list[AnyMessage] | AnyMessage | a2a_types.Message | a2a_types.Artifact
     ) -> AnyMessage:
         if isinstance(input, str):
             return UserMessage(input)
         elif isinstance(input, Message):
             return input
+        elif isinstance(input, list) and input and isinstance(input[-1], Message):
+            return input[-1]
```

Comment:
The only place where we are passing multiple messages is here.

https://github.com/i-am-bee/beeai-framework/blob/82120b5ed41f8eea26ed89dc5e34d3ff496e21f5/python/beeai_framework/adapters/a2a/agents/agent.py#L132-L134

If we used your changes, then we would lose all new messages except the last one.

In my opinion, we should keep the function as is and make this change instead.

```python
input_messages = [msg for msg in self._convert_message_to_framework_message(input)]
await self.memory.add_many(input_messages)
```

Reply:
Thanks. Fixed.